src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -11,50 +11,109 @@
 
 import typing as t
 
-from taipy.core import Core, Scenario
+from dateutil import parser
+
+import taipy as tp
 from taipy.gui import Gui, State
 from taipy.gui.extension import Element, ElementLibrary, ElementProperty, PropertyType
 
 
 class GuiCoreContext:
```

Comment:
Rename to CoreElementLibrary? CoreComponentLibrary?

Reply:
I'll let you decide.
But as it is now automatically registered ...

Reply:
"Component" is reserved for the front-end (React).
"Block" is for structuring elements, "Control" for the others.
The proper naming the the class would be "CoreElementLibrary".